services/src/main/java/org/keycloak/authentication/authenticators/client/X509ClientAuthenticator.java

```diff
@@ -0,0 +1,142 @@
+package org.keycloak.authentication.authenticators.client;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.authentication.AuthenticationFlowError;
+import org.keycloak.authentication.ClientAuthenticationFlowContext;
+import org.keycloak.models.AuthenticationExecutionModel;
+import org.keycloak.models.ClientModel;
+import org.keycloak.protocol.oidc.OIDCLoginProtocol;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.keycloak.services.ServicesLogger;
+import org.keycloak.services.x509.X509ClientCertificateLookup;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import java.security.GeneralSecurityException;
+import java.security.cert.X509Certificate;
+import java.util.*;
+
+public class X509ClientAuthenticator extends AbstractClientAuthenticator {
+
+    public static final String PROVIDER_ID = "client-x509";
```

Comment:
I'm just thinking about this name... it appears in the UI. Maybe we should call it "Client x509 certificate" or "Mutual TLS certificate"?

Reply:
That's the id, I tried to stay consistent with the other authenticator `client-secret` , `client-jwt`

Reply:
Yeah, that's just the id. The display name is returned by "getDisplayType()" and that is the one displayed in the admin console UI

Reply:
Ok, great. Thanks for clarification.